### ISSUE TITLE
UPDATE and DELETE JPA  does not get applied with WHERE clause

### ISSUE THREAD

**Issue by Deepak-Vohra** [Impetus/kundera#517]
The executeUpdate() does not apply UPDATE and DELETE queries if WHERE clause used in query. 

For UPDATE with WHERE the following exception is generated.

EntityManagerFactory created for persistence unit : kundera
Exception in thread "main" java.util.NoSuchElementException

For DELETE with WHERE an error is not generated but a delete is not made. 

Example too extensive to list. Shall send separately if required.

Columns used in WHERE are indexed columns.

**Comment by mevivs** [comment]
Kundera version?
Database name/version?
share JPQL query.

```
For DELETE with WHERE an error is not generated but a delete is not made.
```

Not able to understand what you mean by above statement

-Vivek

**Comment by Deepak-Vohra** [comment]
Kundera version?
2.2.1
Database name/version?
Cassandra 2.04
share JPQL query.

em.createQuery( "UPDATE Catalog c  SET c.author = 'Kelly, David A.'  WHERE c.catalogId='catalog1'") .executeUpdate();

em.createQuery( "DELETE FROM Catalog c WHERE c.title='Engineeering As a Service'").executeUpdate();

For DELETE with WHERE an error is not generated but a delete is not made.

Not able to understand what you mean by above statement

In the preceding statements WHERE is included and the UPDATE/DELETE does not get applied. Without the WHERE clause UPDATE/DELETE gets applied.

**Comment by mevivs** [comment]
Hi,
2.2.1 is quite old release. Latest released version is 2.9.1. Here i am sharing a junit quite similar to what you are trying at your end. Here it is:
https://github.com/impetus-opensource/Kundera/blob/trunk/src/kundera-cassandra/src/test/java/com/impetus/client/crud/UpdateDeleteJPQLLiteralTest.java

Hope it helps.

-Vivek
